contracts/mainnet/connectors/sushi-incentive/main.sol

```diff
@@ -148,24 +169,27 @@ abstract contract SushipswapIncentiveResolver is Helpers, Events {
     function emergencyWithdraw(
         address token1,
         address token2,
-        uint256 setId
+        uint256 setId,
+        Metadata memory data
     )
         external
         payable
         returns (string memory _eventName, bytes memory _eventParam)
     {
         token1 = changeEthAddrToWethAddr(token1);
         token2 = changeEthAddrToWethAddr(token2);
-        (uint256 _pid, uint256 _version, ) = _getPoolId(token1, token2);
-        setUint(setId, _pid);
-        require(_pid != uint256(-1), "pool-does-not-exist");
-        (uint256 lpAmount, uint256 rewardsAmount) = _getUserInfo(
-            _pid,
-            _version
-        );
-        _emergencyWithdraw(_pid, _version);
+        if(data.poolId == uint256(-1)|| data.version <= 0 ){
+            data = _getPoolId(
+                token1,
+                token2
+            );
+        }
+        setUint(setId, data.poolId);
```

Comment:
I don't think we need to set poolId?

Reply:
@pradyuman-verma Maybe we can set withdrawn amount

Reply:
but there is no param as amount in here !

Reply:
I think we can find out using initial and final balance of the token?

Reply:
We're withdrawing all the supplied assets in emergency withdrawal, as we aren't providing how much amount we want to remove?